From 67f85522c7cb52e4a3bf31275ffcf0e2c90c769a Mon Sep 17 00:00:00 2001
From: Anil Kumar Mamidala <anil.mamidala@xilinx.com>
Date: Tue, 6 Jul 2021 12:33:58 +0530
Subject: [PATCH] media: i2c: ap1302: Add a flag to check the streaming status

Add a flag to check the streaming status of ISP and avoid calling
stream on logic multiple times. Updating the registers of ISP when
ISP is in streaming state will cause errors.

Signed-off-by: Anil Kumar Mamidala <anil.mamidala@xilinx.com>
---
 drivers/media/i2c/ap1302.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index b508dd87f0b4..8a28c519c8e5 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -408,6 +408,7 @@ struct ap1302_device {
 	struct media_pad pads[AP1302_PAD_MAX];
 	struct ap1302_format formats[AP1302_PAD_MAX];
 	unsigned int width_factor;
+	bool streaming;
 
 	struct v4l2_ctrl_handler ctrls;
 
@@ -1211,8 +1212,10 @@ static int ap1302_stall(struct ap1302_device *ap1302, bool stall)
 		if (ret < 0)
 			return ret;
 
+		ap1302->streaming = false;
 		return 0;
 	} else {
+		ap1302->streaming = true;
 		return ap1302_write(ap1302, AP1302_SYS_START,
 				    AP1302_SYS_START_PLL_LOCK |
 				    AP1302_SYS_START_STALL_STATUS |
@@ -1670,6 +1673,9 @@ static int ap1302_s_stream(struct v4l2_subdev *sd, int enable)
 
 	mutex_lock(&ap1302->lock);
 
+	if (enable == ap1302->streaming)
+		goto done;
+
 	if (enable) {
 		ret = ap1302_configure(ap1302);
 		if (ret < 0)
-- 
2.17.1

